src/EFCore.PG.NodaTime/Query/ExpressionTranslators/Internal/NpgsqlNodaTimeMemberTranslatorPlugin.cs

```diff
@@ -76,9 +76,49 @@ public NpgsqlNodaTimeMemberTranslator(ISqlExpressionFactory sqlExpressionFactory
                 return TranslateDateTime(instance, member, returnType);
             }
 
+            if (instance is not null && declaringType == typeof(Duration))
+            {
+                return TranslateDuration(instance, member);
+            }
+
             return null;
         }
 
+        /// <summary>
+        /// Translates Duration members
+        /// </summary>
+        /// <param name="instance"></param>
+        /// <param name="member"></param>
+        /// <returns>The translated expression or null</returns>
+        private SqlExpression? TranslateDuration(SqlExpression instance, MemberInfo member)
+        {
+            switch (member.Name)
+            {
+                case nameof(Duration.TotalDays):
+                case nameof(Duration.TotalHours):
+                case nameof(Duration.TotalMinutes):
+                case nameof(Duration.TotalSeconds):
+                    var divisor = member.Name switch
+                    {
+                        nameof(Duration.TotalDays) => 86400,
+                        nameof(Duration.TotalHours) => 3600,
+                        nameof(Duration.TotalMinutes) => 60,
+                        nameof(Duration.TotalSeconds) => 1,
+                        _ => 0 // cannot happen
```

Comment:
If you decide to keep this inner switch (see comment above), better to throw an exception here instead.